src/graph_impl/mod.rs

```diff
@@ -854,6 +854,24 @@ impl<N, E, Ty, Ix> Graph<N, E, Ty, Ix>
         }
     }
 
+    /// Return an iterator over all edges connecting `a` and `b`.
+    ///
+    /// - `Directed` and `Undirected`: All edges connecting `a` and `b`.
+    ///
+    /// Iterator element type is `EdgeReference<E, Ix>`.
+    pub fn edges_connecting(&self, a: NodeIndex<Ix>, b: NodeIndex<Ix>) -> MultipleEdges<E, Ty, Ix>
```

Comment:
Small style note. The iterator naming guideline is that it is almost always named after the method that creates it (we can test this in libstd or in petgraph).

Reply:
I'd expect this method to return the edges connecting with directionality, that is from *a* to *b* for a directed graph?

Either way, why not write it as a filter for the Edges iterator, that way all the special cases are already correctly handled (mainly directionality and self loops).

Reply:
Thanks for the review! I just addressed your comments in a couple of new commits